staging/src/k8s.io/client-go/example_test.go

```diff
@@ -0,0 +1,429 @@
+		fmt.Printf("Error encountered: %v\n", err)
+		return
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Printf("Error encountered: %v\n", err)
+		return
+	}
+
+	// Define the desired state of the Deployment using the applyconfigurations package.
+	// This provides a typed, structured way to build the patch.
+	deploymentName := "my-app"
+	replicas := int32(2)
+	image := "nginx:1.14.2"
+
+	// The FieldManager is a required field that identifies the controller managing
+	// this object's state.
+	fieldManager := "my-controller"
+
+	// Build the apply configuration.
+	deploymentApplyConfig := appsv1ac.Deployment(deploymentName, "default").
+		WithSpec(appsv1ac.DeploymentSpec().
+			WithReplicas(replicas).
+			WithSelector(metav1ac.LabelSelector().WithMatchLabels(map[string]string{"app": "my-app"})).
+			WithTemplate(corev1ac.PodTemplateSpec().
+				WithLabels(map[string]string{"app": "my-app"}).
+				WithSpec(corev1ac.PodSpec().
+					WithContainers(corev1ac.Container().
+						WithName("nginx").
+						WithImage(image),
+					),
+				),
+			),
+		)
+
+	// Perform the Server-Side Apply patch. The PatchType must be types.ApplyPatchType.
+	// The context, name, apply configuration, and patch options are required.
+	result, err := clientset.AppsV1().Deployments("default").Apply(
+		context.TODO(),
+		deploymentApplyConfig,
+		metav1.ApplyOptions{FieldManager: fieldManager},
+	)
+
+	if err != nil {
+		fmt.Printf("Error encountered: %v\n", err)
+		return
+	}
+
+	fmt.Printf("Deployment %q applied successfully.\n", result.Name)
+}
+
+func Example_leaderElection() {
+	// This example demonstrates the leader election pattern. A controller running
+	// multiple replicas uses leader election to ensure that only one replica is
+	// active at a time.
+
+	// Configure the client (out-of-cluster for this example).
+	var kubeconfig string
+	if home := os.Getenv("HOME"); home != "" {
+		kubeconfig = filepath.Join(home, ".kube", "config")
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		fmt.Printf("Error building kubeconfig: %s\n", err.Error())
+		return
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Printf("Error building clientset: %s\n", err.Error())
+		return
+	}
+
+	// The unique name of the controller writing to the Lease object.
+	id := "my-controller"
+
+	// The namespace and name of the Lease object.
+	leaseNamespace := "default"
+	leaseName := "my-controller-lease"
+
+	// Create a context that can be cancelled to stop the leader election.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Set up a signal handler to cancel the context on termination.
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-sigCh
+		log.Println("Received termination signal, shutting down...")
+		cancel()
+	}()
+
+	// Create the lock object.
+	lock := &resourcelock.LeaseLock{
```

Comment:
nit: Usually I've seen `k8s.io/client-go/tools/leaderelection/resourcelock.New()` to create this rather directly creating the object.